src/main/java/de/tum/in/www1/artemis/web/rest/programming/ProgrammingExerciseExportImportResource.java

```diff
@@ -529,4 +529,32 @@ public ResponseEntity<Resource> exportStudentRequestedRepository(@PathVariable l
 
         return returnZipFileForRepositoryExport(zipFile, RepositoryType.SOLUTION.getName(), programmingExercise, start);
     }
+
+    /**
+     * GET /programming-exercises/:exerciseId/export-student-repository/:participationId : Exports the repository belonging to a participation as a zip file.
+     *
+     * @param exerciseId      The id of the programming exercise
+     * @param participationId The id of the student participation for which to export the repository.
+     * @return A ResponseEntity containing the zipped repository.
+     * @throws IOException If the repository could not be zipped.
+     */
+    @GetMapping("programming-exercises/{exerciseId}/export-student-repository/{participationId}")
+    @EnforceAtLeastStudent
+    @FeatureToggle({ Feature.ProgrammingExercises, Feature.Exports })
```

Comment:
Why do we add this annotation to every method instead of the class here?

Reply:
Interestingly, I've only ever seen this as a method-level annotation. I wonder what the reason for it is. I'd say it's worth exploring in a separate PR as it seems to affect multiple Resource classes